conda_pack/formats.py

```diff
@@ -13,244 +13,35 @@
-                    if len(buf):
-                        self.fileobj.write(buf)
-
-    def _compress(self, in_bufs):
-        out_bufs = []
-        compressor = self._new_compressor()
-        for data in in_bufs:
-            out_bufs.append(compressor.compress(data))
-        out_bufs.append(self._flush_compressor(compressor))
-        return out_bufs
-
-    def close(self):
-        if self.fileobj is None:
-            return
-
-        # Flush any waiting buffers
-        if self.buffers:
-            self.compress_queue.put(self.buffers)
-
-        # Wait for all work to finish
-        self.compress_queue.put(None)
-        self._consumer_thread.join()
-
-        # Write the closing bytes
-        self._write_footer()
-
-        # Flush fileobj
-        self.fileobj.flush()
-
-        # Cache shutdown state
-        self.compress_queue = None
-        self.pool = None
-        self.fileobj = None
-
-
-class ParallelGzipFileWriter(ParallelFileWriter):
-    # Since it's hard for us to keep a running dictionary (a serial operation)
-    # with parallel compression of blocks, we use a blocksize > a few factors
-    # bigger than the max dict size (32 KiB). In practice this is fine - we
-    # only lose out by a small factor of unneeded redundancy, and real files
-    # often lack enough redundant byte sequences to make this significant. Pigz
-    # uses 128 KiB, but does more work to keep a running dict.
-    _block_size = 256 * 2**10
-
-    def _init_state(self):
-        self.crc = zlib.crc32(b"") & 0xffffffff
-
-    def _new_compressor(self):
-        return zlib.compressobj(self.compresslevel, zlib.DEFLATED,
-                                -zlib.MAX_WBITS, zlib.DEF_MEM_LEVEL, 0)
-
-    def _per_buffer_op(self, buffer):
-        self.crc = zlib.crc32(buffer, self.crc) & 0xffffffff
-
-    def _write32u(self, value):
-        self.fileobj.write(struct.pack("<L", value))
-
-    def _write_header(self):
-        self.fileobj.write(b'\037\213\010')
-        self.fileobj.write(b'\x00')
-        self._write32u(int(time.time()))
-        self.fileobj.write(b'\002\377')
-
-    def _write_footer(self):
-        self.fileobj.write(self._new_compressor().flush(zlib.Z_FINISH))
-        self._write32u(self.crc)
-        self._write32u(self.size & 0xffffffff)
-
-    def _flush_compressor(self, compressor):
-        return compressor.flush(zlib.Z_FULL_FLUSH)
-
-
-class ParallelBZ2FileWriter(ParallelFileWriter):
-    def _init_state(self):
-        # bzip2 compresslevel dictates its blocksize of 100 - 900 kb
-        self._block_size = self.compresslevel * 100 * 2**10
-
-    def _new_compressor(self):
-        import bz2
-        return bz2.BZ2Compressor(self.compresslevel)
-
-    def _per_buffer_op(self, buffer):
-        pass
-
-    def _write_header(self):
-        pass
-
-    def _write_footer(self):
-        pass
-
-    def _flush_compressor(self, compressor):
-        return compressor.flush()
-
-
-class ParallelXZFileWriter(ParallelFileWriter):
```

Comment:
these will need to be preserved; or we could simplify and make you use zstd if you need speed